Scheduled prompts

adamsullovey opened the pull request:
This PR adds scheduling system that can be used to have penny send scheduled prompts to the chat room. The messages & schedule are hardcoded right now. I'm sure that will change eventually.

closes #12

Commit:
Add scheduled prompts to penny

I put this code in a class so that it's easy to to test or have multiple
prompts scheduled. I expect SimplePromptGenerator will be replaced
eventually, but it is helpful for now.

This successfully put a message in the #draw_it channel at today at
2:27pm.

Commit:
Stop sending scheduled prompts when offline

This will prevent some errors from appearing in the bot's console.

adamsullovey commented:
I have changes from the-heap:master merged into my master adamsullovey:master. Does it matter if I merge or rebase them into this branch (adamsullovey:scheduled-prompts) for this PR?

teesloane commented:
I don't think it should matter for you to merge/rebase. If it's too much trouble, I can do it from the CLI :)

Will review this at some point today. Excited.

Commit:
Send scheduled prompt once an hour

because once a minute is too much chatting when many people are testing
the bot.

Commit:
Merge branch 'master' into scheduled-prompts

Commit:
Move .js files to src folder

to match the new directory structure in upstream/master
